PowerEditor/src/NppIO.cpp

```diff
@@ -1194,6 +1194,11 @@ bool Notepad_plus::fileSave(BufferID id)
 				}
 			}
 
+			// Expand any environment variables
+			TCHAR fn_bak_expanded[MAX_PATH] = { '\0' };
+			::ExpandEnvironmentStrings(fn_bak.c_str(), fn_bak_expanded, MAX_PATH);

```

Comment:
Might be wise to check the return value, see [ExpandEnvironmentStrings help](https://msdn.microsoft.com/en-us/library/windows/desktop/ms724265(v=vs.85).aspx) and it's [example](https://msdn.microsoft.com/en-us/library/windows/desktop/ms724426(v=vs.85).aspx).